connectivity/tests/host.go

```diff
@@ -35,7 +35,15 @@ func (s *podToHost) Run(ctx context.Context, t *check.Test) {
 		for _, node := range ct.Nodes() {
 			node := node
 
+			var ipFamStr string
+
 			t.ForEachIPFamily(func(ipFam check.IPFamily) {
+				if ipFam == check.IPFamilyV4 {
+					ipFamStr = "v4"
+				} else if ipFam == check.IPFamilyV6 {
+					ipFamStr = "v6"
+				}
```

Comment:
Thanks! Golang does not support the ternary operator, that's why you get syntax errors.

What I would recommend instead however is either move this into a `ShortString()` method, or (my preference) just use the existing String() method. It will print `ipv4` instead of `v4`, but I think that's actually fine. Using `String()`, you can just pass the IPFamily directly to `Sprintf` with the `%s` specifier, no manual conversion to a string needed.

Reply:
@gandro Gotcha! So, it should be like this, right? 👇🏼 

```go
func (s *podToHost) Run(ctx context.Context, t *check.Test) {
	ct := t.Context()
	// Construct a unique list of all nodes in the cluster running workloads.

	var i int

	for _, pod := range ct.ClientPods() {
		pod := pod // copy to avoid memory aliasing when using reference

		for _, node := range ct.Nodes() {
			node := node

			t.ForEachIPFamily(func(ipFam check.IPFamily) {
				dstAddrs := []string{}
				for _, addr := range node.Status.Addresses {
					if check.GetIPFamily(addr.Address) != ipFam {
						continue
					}
					dstAddrs = append(dstAddrs, addr.Address)
				}
				if len(dstAddrs) == 0 {
					return
				}

				dst := check.ICMPEndpoint("", dstAddrs...)
				t.NewAction(s, fmt.Sprintf("ping-%s-%d", ipFam, i), &pod, dst, ipFam).Run(func(a *check.Action) {
					a.ExecInPod(ctx, ct.PingCommand(dst, ipFam))

					a.ValidateFlows(ctx, pod, a.GetEgressRequirements(check.FlowParameters{
						Protocol: check.ICMP,
					}))
				})

				i++
			})
		}
	}
}
```

If yes, then I got the rest of the part.

Reply:
Yes, I think `fmt.Sprintf("ping-%s-%d", ipFam, i)` looks correct to me.